test/integration/node-specific/mongo_client.test.ts

```diff
@@ -722,6 +722,22 @@ describe('class MongoClient', function () {
         expect(client.s.sessionPool.sessions).to.have.lengthOf(1);
       });
     });
+
+    context('concurrent calls', () => {
+      context('when two concurrent calls to close() occur', () => {
+        it('no error is thrown', async function () {
+          expect(() => Promise.all([client.close(), client.close()])).to.not.throw;
+        });
+      });
+
+      context('when more than two concurrent calls to close() occur', () => {
+        it('no error is thrown', async function () {
+          expect(() =>
+            Promise.all([client.close(), client.close(), client.close(), client.close()])
+          ).to.not.throw;
+        });
+      });
```

Comment:
The client.connect() tests also ensure that if an error is thrown, `connectionLock` is unset after the promise rejects.  Can we add a similar test for close()?

Reply:
yep!